src/extension.ts

```diff
@@ -25,9 +48,38 @@ export async function activate(context: vscode.ExtensionContext) {
   context.subscriptions.push(git.onDidOpenRepository(() => checkEnabled(git)));
   context.subscriptions.push(git.onDidCloseRepository(() => checkEnabled(git)));
 
+  // Create a debounced version of updateStatusBarItem
+  let updateTimeout: NodeJS.Timeout | null = null;
+  const debouncedUpdateStatusBar = (editor: vscode.TextEditor | undefined) => {
```

Comment:
Do we actually need this denounced handler? I don't imagine that a user could change config settings, or their active file, faster than 50ms. At which point, this might be an unnecessary guardrail?